node/network/gossip-support/src/lib.rs

```diff
@@ -212,7 +223,14 @@ where
 				}
 
 				let all_authorities = determine_relevant_authorities(ctx, relay_parent).await?;
-				let our_index = ensure_i_am_an_authority(&self.keystore, &all_authorities).await?;
+				let our_index = ensure_i_am_an_authority(&self.keystore, &all_authorities)
+					.await
+					.map_err(|e| {
+						self.metrics.on_is_not_authority();
```

Comment:
I find side effects in `map` functions rather ugly - I think an explicit pattern match would be better and more readable, also hardly more code.

The other thing is, as @ordian mentioned - this won't do exactly what you want as `ensure_i_am_an_authority` will check whether we are an authority in a couple of sessions, not just the current one.

Reply:
> I find side effects in `map` functions rather ugly - I think an explicit pattern match would be better and more readable, also hardly more code.
I agree in the general case, but in this particular one there are no side effects, it's a metric increment which is on the same level with logging wrt side effects (state). Initially I went for the match, but I like it better this way. If you have a hard preference here I'll switch to match.

> The other thing is, as @ordian mentioned - this won't do exactly what you want as `ensure_i_am_an_authority` will check whether we are an authority in a couple of sessions, not just the current one.

Yeah, that makes sense, but my thinking was this doesn't need to be very accurate so, assuming couple of sessions is not hours it might be OK. @ordian @eskimor how many blocks does this translate to currently ?